trade-finance-manager-ui/server/controllers/utilisation-reports/record-corrections/create-record-correction-request/index.test.ts

```diff
@@ -42,25 +47,182 @@ describe('controllers/utilisation-reports/record-corrections/create-record-corre
-        facilityId: '0012345678',
-        exporter: 'Sample Company Ltd',
-      },
-      backLinkHref: getLinkToPremiumPaymentsTab(reportId, [456]),
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  describe('getCreateRecordCorrectionRequest', () => {
+    const getHttpMocks = () =>
+      httpMocks.createMocks<GetCreateRecordCorrectionRequestRequest>({
+        params: { reportId, feeRecordId },
+        session: requestSession,
+      });
+
+    it('should render the create record correction request page', async () => {
+      // Arrange
+      const { req, res } = getHttpMocks();
+
+      // Act
+      await getCreateRecordCorrectionRequest(req, res);
+
+      // Assert
+      expect(res._getRenderView()).toEqual('utilisation-reports/record-corrections/create-record-correction-request.njk');
+      expect(res._getRenderData() as CreateRecordCorrectionRequestViewModel).toEqual<CreateRecordCorrectionRequestViewModel>({
+        bank: { name: 'Test Bank' },
+        user,
+        activePrimaryNavigation: PRIMARY_NAVIGATION_KEYS.UTILISATION_REPORTS,
+        reportId,
+        formattedReportPeriod: 'January 2024',
+        feeRecord: {
+          facilityId: '0012345678',
+          exporter: 'Sample Company Ltd',
+        },
+        formValues: {},
+        errors: { errorSummary: [] },
+        backLinkHref: getLinkToPremiumPaymentsTab(reportId, [456]),
+      });
+
+      expect(api.getFeeRecord).toHaveBeenCalledTimes(1);
+      expect(api.getFeeRecord).toHaveBeenCalledWith(reportId, feeRecordId, userToken);
+    });
+  });
+
+  describe('postCreateRecordCorrectionRequest', () => {
+    beforeEach(() => {
+      jest.mocked(validateCreateRecordCorrectionRequestFormValues).mockReturnValue(EMPTY_CREATE_RECORD_CORRECTION_REQUEST_ERRORS_VIEW_MODEL);
+    });
+
+    afterEach(() => {
+      jest.resetAllMocks();
+    });
+
+    describe('when the form values are valid', () => {
+      const validBody = aPostCreateRecordCorrectionRequestBody();
+      const getHttpMocks = () =>
+        httpMocks.createMocks<PostCreateRecordCorrectionRequestRequest>({
+          params: { reportId, feeRecordId },
+          session: requestSession,
+          body: validBody,
+        });
+
+      it('redirects to the "check the information" page', async () => {
+        // Arrange
+        const { req, res } = getHttpMocks();
+
+        // Act
+        await postCreateRecordCorrectionRequest(req, res);
+
+        // Assert
+        expect(res._getRedirectUrl()).toEqual(`/utilisation-reports/${reportId}/create-record-correction-request/${feeRecordId}/check-the-information`);
+        expect(api.getFeeRecord).not.toHaveBeenCalled();
+
+        expect(validateCreateRecordCorrectionRequestFormValues).toHaveBeenCalledTimes(1);
+        expect(validateCreateRecordCorrectionRequestFormValues).toHaveBeenCalledWith(validBody);
+      });
+
+      function aPostCreateRecordCorrectionRequestBody(): CreateRecordCorrectionRequestFormRequestBody {
```

Comment:
Can we just have this in the `validBody` const? I think this function is unnecessary as it's only used in one place